sycl/unittests/scheduler/InOrderQueueHostTaskDeps.cpp

```diff
@@ -52,3 +53,43 @@ TEST_F(SchedulerTest, InOrderQueueHostTaskDeps) {
 
   EXPECT_TRUE(GEventsWaitCounter == 1);
 }
+
+constexpr auto DisablePostEnqueueCleanupName =
+    "SYCL_DISABLE_POST_ENQUEUE_CLEANUP";
+
+TEST_F(SchedulerTest, InOrderQueueSubmissionOrder) {
```

Comment:
The exact intention of this test is less clear. Could you please add either comment here explaining it or add some comments throughout the code to explain the different parts?

Reply:
Hi @steffenlarsen for now I would like to check strict order of enqueue in case of inorder queue. One of my patches broke this that caused kernel2 to be enqueued earlier.